c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1339,8 +1339,7 @@ class THCCachingAllocator {
     if (size < device_count) {
       device_allocator.resize(device_count);
```

Comment:
Given that these elements are not used, it'd be nice to reserve rather than resize
```c++
device_allocator.reserve(device_count);
for (const auto _ : c10::irange(device_count - size)) {
  device_allocator.push_back(std::make_unique<DeviceCachingAllocator>());
```